src/actions/sagemaker/sagemaker_infer.ts

```diff
@@ -0,0 +1,356 @@
+/* tslint:disable max-line-length */
+
+import * as Hub from "../../hub"
+
+import * as S3 from "aws-sdk/clients/s3"
+import * as SageMaker from "aws-sdk/clients/sagemaker"
+import { PassThrough } from "stream"
+import * as winston from "winston"
+import { awsInstanceTypes } from "./aws_instance_types"
+
+const stripLines = require("striplines")
+const stripColumns = require("./strip_columns.js")
+
+function logJson(label: string, obj: any) {
+  winston.debug(label, JSON.stringify(obj, null, 2))
+}
+
+export class SageMakerInferAction extends Hub.Action {
+
+  name = "amazon_sagemaker_infer"
+  label = "Amazon SageMaker Infer"
+  iconName = "sagemaker/sagemaker_infer.png"
+  description = "Perform an inference using Amazon SageMaker."
+  supportedActionTypes = [Hub.ActionType.Query]
+  supportedFormats = [Hub.ActionFormat.Csv]
+  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+  supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+
+  usesStreaming = true
+  requiredFields = []
+
+  params = [
+    {
+      name: "accessKeyId",
+      label: "Access Key",
+      required: true,
+      sensitive: true,
+      description: "Your access key for SageMaker.",
+    },
+    {
+      name: "secretAccessKey",
+      label: "Secret Key",
+      required: true,
+      sensitive: true,
+      description: "Your secret key for SageMaker.",
+    },
+    {
+      name: "roleArn",
+      label: "Role ARN",
+      required: true,
+      sensitive: false,
+      description: "Role ARN for accessing SageMaker and S3",
+    },
+    {
+      name: "region",
+      label: "Region",
+      required: true,
+      sensitive: false,
+      description: "AWS Region for accessing SageMaker",
+    },
+    {
+      name: "user_email",
+      label: "Looker User Email",
+      required: true,
+      description: `
+        Click the button on the right and select 'Email'.
+        This is required for the action to send status emails
+        when training or inference jobs are complete.
+      `,
+      sensitive: false,
+    },
+    {
+      name: "smtpHost",
+      label: "SMTP Host",
+      required: true,
+      sensitive: false,
+      description: "Host for sending emails.",
+    },
+    {
+      name: "smtpPort",
+      label: "SMTP Port",
+      required: true,
+      sensitive: false,
+      description: "Port for sending emails.",
+    },
+    {
+      name: "smtpFrom",
+      label: "SMTP From",
+      required: true,
+      sensitive: false,
+      description: "From for sending emails.",
+    },
+    {
+      name: "smtpUser",
+      label: "SMTP User",
+      required: true,
+      sensitive: false,
+      description: "User for sending emails.",
+    },
+    {
+      name: "smtpPass",
+      label: "SMTP Pass",
+      required: true,
+      sensitive: false,
+      description: "Pass for sending emails.",
+    },
+  ]
+
+  async execute(request: Hub.ActionRequest) {
+
+    logJson("request", request)
```

Comment:
I haven't noticed many debug statements in the core repo, should these be removed before a PR to master?

Reply:
paring out most of these debug statements